js/map.js

```diff
@@ -1,171 +1,129 @@
-class GooleMapsMap {
-  constructor(apiUrl, events, getCountryCode, options) {
-    this.events = events;
-    this.GEOCODE_API = 'https://maps.googleapis.com/maps/api/geocode/json?address=';
-    // add google vendor scripts
-    this.loadJS(apiUrl);
-    // spread sheet keys
-    this.CITY_KEY = 'City';
-    this.COUNTRY_KEY = 'Country';
-    this.LAT_KEY = 'Lat';
-    this.LNG_KEY = 'Lng';
-    this.REGISTRATION_KEY = 'RegistrationLink';
-    this.HOST_KEY = 'FullName';
-    this.DATE_KEY = 'Date';
-    this.MARKER_ICON = 'https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%E2%80%A2%7C6FB8D8';
-    this.getCountryCode = getCountryCode;
-    this.linkCountry = options.hasContainerEventList;
-    this.hasEventDetail = options.hasContainerEventDetail;
-    this.markers = [];
-  }
+'use strict';
 
-  loadJS(file) {
-    const jsElement = document.createElement("script");
-    jsElement.type = "application/javascript";
-    jsElement.src = file;
-    document.body.appendChild(jsElement);
-  }
+var _createClass = function () { 
```

Comment:
this is still very confusing to me. Can we just use JavaScript prototypes? https://code.tutsplus.com/tutorials/prototypes-in-javascript--net-24949

Reply:
never was friendly with objects. :disappointed: While haven't idea how to make it work.